src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -32,6 +29,13 @@ end
 # Tree adaptations of functionalities in ITensors.jl/src/physics/autompo/opsum_to_mpo.jl
 # 
 
+function determine_val_type(terms)
```

Comment:
```suggestion
function promote_coefficient_type(terms)
```
Also I think we can write this logic in a better way, for example:
```julia
function promote_coefficient_type(terms)
  return mapreduce(t -> typeof(coefficient(t)), promote_type, terms)
end
```

Reply:
Interestingly, this is a code improvement that reveals another possible bug, this time in `OpSum` or `Ops`. Benedikt noted this bug in a comment in the code too. Here's the bug:
```julia
using ITensors: OpSum
let
  terms = OpSum()
  terms .+= 1.0, "Sz",1,"Sz",2
  @show typeof(coefficient(first(terms)))
  #
  # Prints:
  #   typeof(coefficient(first(terms))) = ComplexF64
  #
  return
end
```
The `typeof(coefficient(t))` for any term in an `OpSum` is `ComplexFloat64` regardless of whether the original coefficient is real or not.

Reply:
That's by design, `OpSum()` is defined as `OpSum{ComplexF64}()`.

That was to allow backwards compatibility with the old `OpSum`/`AutoMPO` type, otherwise:
```julia
  terms = OpSum()
  terms .+= 1.0 + 2.0im, "Sz",1,"Sz",2
```
would error.

If you want to check if the coefficients are numerically real (i.e. `Float64`, or `ComplexF64` with zero imaginary part) you would have to use a runtime check like `isreal` rather than check the type, or constrain the `OpSum` to be real with `OpSum{Float64}()`.

Reply:
Good to know – I had forgotten if it was by design or not. I'll come up with a similar function to `promote_coefficient_type` above, then, that uses a more generic and modern style.